src/classes/Listings/Listings.ts

```diff
@@ -57,26 +59,19 @@ export default class Listings {
             return;
         }
 
-        this.isForceDisabled = false;
-
         // Autobump is enabled, add heartbeat listener
 
-        this.bot.listingManager.removeListener('heartbeat', this.checkAccountInfo.bind(this));
-        this.bot.listingManager.on('heartbeat', this.checkAccountInfo.bind(this));
+        this.bot.listingManager.removeListener('heartbeat', this.checkFn);
+        this.bot.listingManager.on('heartbeat', this.checkFn);
 
         // Get account info
         this.checkAccountInfo();
     }
 
     disableAutorelistOption(): void {
-        if (!this.autoRelistEnabled) {
-            // If the process already not running, just don't execute this.
-            return;
-        }
-        this.bot.listingManager.removeListener('heartbeat', this.checkAccountInfo.bind(this));
-        this.bot.listingManager.off('heartbeat', this.checkAccountInfo.bind(this));
-        this.isForceDisabled = true;
-        this.disableAutoRelist();
```

Comment:
Thanks, but I wonder why you remove this and replace it with the same thing inside this function.

Reply:
I mean, `this.disableAutoRelist();`
https://github.com/TF2Autobot/tf2autobot/blob/58f6ee85c5ffb0d1c7a0d5b71115ba2088666b0a/src/classes/Listings/Listings.ts#L150-L155

Reply:
I think I was just leaving it how it was prior to the attempted fixes. I cleaned it up and put it through disableAutoRelist now